xDrip/Scenes/Entries/EditTraining/EditTrainingViewController.swift

```diff
@@ -59,13 +63,95 @@ class EditTrainingViewController: NibViewController, EditTrainingDisplayLogic {
     // MARK: Do something
     
     private func doLoad() {
+        setupNavigationItems()
+        setupTableView()
+        
         let request = EditTraining.Load.Request()
         interactor?.doLoad(request: request)
     }
     
+    private func setupNavigationItems() {
+        var rightItem = UIBarButtonItem.SystemItem.done
+        
+        switch dataStore?.mode {
```

Comment:
accessing dataStore from view controller is not a good approach. Pass it to view controller in `Load` event.